src/NHibernate/Linq/NestedSelects/NestedSelectRewriter.cs

```diff
@@ -16,7 +16,18 @@ namespace NHibernate.Linq.NestedSelects
 {
 	static class NestedSelectRewriter
 	{
-		private static readonly MethodInfo CastMethod = EnumerableHelper.GetMethod("Cast", new[] { typeof (IEnumerable) }, new[] { typeof (object[]) });
+		private static readonly MethodInfo CastMethod = ReflectionHelper.GetMethod(
```

Comment:
I have to check if a sound naming convention would do. With overloads and methods having the same name but belonging to other types, putting them all on `ReflectionHelper` could end up a bit messy.
Maybe should we use some `ReflectionHelper` internal subclass for each commonly reflected type to hold their cached method definition. I will test that soon.